### ISSUE TITLE
Microsoft.Azure.WebJobs.Script.Grpc: Object reference not set to an instance of an object

### ISSUE THREAD

**Issue by WhyDoYouDoThis** [Azure/azure-functions-dotnet-worker#359]
I have a very simple Azure Function that has been written in Core 3.1 and moved to .net 5.  The Function does nothing more than read the body of the message and do an HttpClient call to another function.  There is currently NO SECURITY, it is anonymous.  The host is run from the cmd line with _**func start --dotnet-isolated --port 8081 --cors * --verbose**_.

While the host runs, builds the end-point, shows it, and seems fine, after a few seconds an exception is displayed in red

**_[2021-03-25T15:49:28.475Z] Exception encountered while listening to EventStream
[2021-03-25T15:49:28.479Z] Microsoft.Azure.WebJobs.Script.Grpc: Object reference not set to an instance of an object._**

My Project file is attached

![image](https://user-images.githubusercontent.com/42220748/112514289-26f80800-8d63-11eb-9e5d-0f6de1df20c6.png)

Program.cs is as follows:

```
_using System;
using System.Diagnostics;
using System.Threading.Tasks;
using Microsoft.Extensions.DependencyInjection;
using Microsoft.Extensions.Hosting;

namespace blahblahblah
{
    class Program
    {
        static void  Main(string[] args)
        {
            Debugger.Launch();
            var host = new HostBuilder()
                .ConfigureFunctionsWorkerDefaults()
                .Build();

            host.Run();
        }
    }
}_

```

**Comment by WhyDoYouDoThis** [comment]
I have updated to the latest bits, including Visual Studio 2019(16.9.2) - still getting the error, still unable to debug.

**Comment by WhyDoYouDoThis** [comment]
So after trying everything I could, I decided to simply download the massive sample repo and start trimming things off until I got to a simple HELLO WORLD. As it is, the code in the sample includes "preview" libraries even though it is built against real bits. It is perfectly okay to update those libraries to the released version. Aside from all the configuration, emulation and what not that is required to get that behemoth "Sample" to work, I have paired it down, updated the libraries and built a working, debuggable "Health Check 200 Ok" sample and made it available here to anyone who wants a boilerplate solution to build from. It is a single project, and only a few nuget packages.

https://drive.google.com/file/d/1-zXQrlxLENAAZ0Fxk0R9s1XCTNxM8p0D/view?usp=sharing
